Hazelcast.Net/Hazelcast.Client.Proxy/TransactionContextProxy.cs

```diff
@@ -22,48 +22,49 @@ namespace Hazelcast.Client.Proxy
 {
     internal class TransactionContextProxy : ITransactionContext
     {
-        internal const int TxOwnerNodeTryCount = 5;
-
         private readonly IDictionary<TransactionalObjectKey, ITransactionalObject> _txnObjectMap =
             new Dictionary<TransactionalObjectKey, ITransactionalObject>(2);
 
-        internal readonly HazelcastClient Client;
-        internal readonly TransactionProxy Transaction;
+        private readonly HazelcastClient _client;
+        private readonly TransactionProxy _transaction;
 
-        internal IMember TxnOwnerNode;
+        internal readonly IMember TxnOwnerNode;
 
         public TransactionContextProxy(HazelcastClient client, TransactionOptions options)
         {
-            Client = client;
-
+            _client = client;
             var clusterService = (ClientClusterService) client.GetClientClusterService();
-            TxnOwnerNode = clusterService.GetRandomMember();
+
+            TxnOwnerNode = client.GetClientConfig().GetNetworkConfig().IsSmartRouting() ?
```

Comment:
Java uses connection for the transaction but .Net seems to use member.  Which one is preferable or more correct?

Reply:
That's a discussion beyond this PR :)

Reply:
This will be a behaviour difference between Java and .Net. Java destroys the transactions when connection is closed but .Net does InvokeOnMember for the txn requests and if a new connection to the member, it will continue.

Reply:
On server side if connection is closed it also removes all transactions. Any new connection cannot recover these transactions so .net client will not continue but instead it will receive a ` TransactionException("No transaction context found for txnId:" + txnId)` from server.

Although there is a difference in handling the closed connection,  there isn't any risk for transactional behaviour.